src/app/clusters/commodity-price-server/commodity-price-server.cpp

```diff
@@ -128,24 +161,326 @@ void Instance::InvokeCommand(HandlerContext & handlerContext)
 {
+    chip::Platform::ScopedMemoryBuffer<Structs::CommodityPriceStruct::Type> forecastBuffer;
+    DataModel::List<const Structs::CommodityPriceStruct::Type> forecastList;
+
     if (!HasFeature(Feature::kForecasting))
     {
         ChipLogError(Zcl, "GetDetailedForecastRequest not supported");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        return;
+    }
+
+    chip::BitMask<CommodityPriceDetailBitmap> details = commandData.details;
+    if (!details.HasOnly(
+            BitMask<CommodityPriceDetailBitmap>(CommodityPriceDetailBitmap::kDescription, CommodityPriceDetailBitmap::kComponents)))
+    {
+        ChipLogError(Zcl, "Invalid details bitmap recvd in GetDetailedForecastRequest");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        return;
+    }
+
+    if (!forecastBuffer.Calloc(kMaxForecastEntries))
+    {
+        ChipLogError(AppServer, "Memory allocation failed for forecast buffer");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ResourceExhausted);
+        return;
+    }
+
+    Commands::GetDetailedForecastResponse::Type response;
+    CHIP_ERROR err = GetDetailedForecastRequest(details, forecastBuffer, kMaxForecastEntries, forecastList, false, true);
+    if (err != CHIP_NO_ERROR)
+    {
         ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
         return;
     }
+    else
+    {
+        response.priceForecast = forecastList;
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+    }
+}
+
+CHIP_ERROR
+Instance::GetDetailedPriceRequest(chip::BitMask<CommodityPriceDetailBitmap> details,
+                                  DataModel::Nullable<Structs::CommodityPriceStruct::Type> & priceStruct)
+{
+
+    if (mCurrentPrice.IsNull())
+    {
+        priceStruct.SetNull();
+        return CHIP_NO_ERROR;
+    }
+    else
+    {
+        priceStruct = mCurrentPrice;
+        if (!details.Has(CommodityPriceDetailBitmap::kComponents))
+        {
+            // Remove the components
+            priceStruct.Value().components.ClearValue();
+        }
+
+        if (!details.Has(CommodityPriceDetailBitmap::kDescription))
+        {
+            // Remove the description
+            priceStruct.Value().description.ClearValue();
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR
+Instance::GetDetailedForecastRequest(chip::BitMask<CommodityPriceDetailBitmap> details,
+                                     chip::Platform::ScopedMemoryBuffer<Structs::CommodityPriceStruct::Type> & forecastBuffer,
+                                     size_t bufferSize, DataModel::List<const Structs::CommodityPriceStruct::Type> & forecastList,
+                                     bool isEvent, bool isCommand)
+{
+
+    size_t count = 0;
+
+    // TODO work out what the max udp packet size is
+    // For events this needs to be about 500
+    // Without events it can be 900
```

Comment:
This is broken for the attribute, no?  For the attribute you want the whole list, not limiting it to one packet, because the whole point is that the attribute can be chunked.

The question is: what should happen for the command?  Spec does not allow doing this truncation business, so what is it expecting to happen, exactly?  Especially since it's providing _more_ information than the attribute, but can't handle chunking....